src/main/java/org/folio/service/processing/reader/SourceReaderBuilder.java

```diff
@@ -40,16 +38,15 @@ public static SourceReader build(File file, JobProfileInfo jobProfile) {
       } else {
         sourceReader = new MarcRawReader(file, MARC_RAW_CHUNK_SIZE);
       }
-    } else if (isEdifact(extension, jobProfile)) {
+    } else if (isEdifact(jobProfile)) {
       sourceReader = new EdifactReader(file, EDIFACT_CHUNK_SIZE);
     }
 
     return ofNullable(sourceReader).orElseThrow(() -> new UnsupportedOperationException("Unsupported file format"));
   }
 
-  private static boolean isEdifact(String extension, JobProfileInfo jobProfile) {
-    return ((jobProfile.getDataType() == JobProfileInfo.DataType.EDIFACT)
-      && (EDIFACT_EDI_EXTENSION.equalsIgnoreCase(extension) || EDIFACT_INV_EXTENSION.equalsIgnoreCase(extension)));
+  private static boolean isEdifact(JobProfileInfo jobProfile) {
+    return jobProfile != null && (jobProfile.getDataType() == JobProfileInfo.DataType.EDIFACT);
```

Comment:
You can use Objects.nonNull method. The static method is doing code more readable.

Reply:
updated